PSTenable/Public/Get-PSTenableSeverity.ps1

```diff
@@ -26,66 +26,93 @@ function Get-PSTenableSeverity {
-                "endOffset"    = 5000
-                "filters"      = [array]@{
-                    "id"           = "severity"
-                    "filterName"   = "severity"
-                    "operator"     = "="
+        $APIresults = @()
+        $CurrentStartOffset = 0
+        
+
+
+        Do {
+            #we're gonna paginate at 5000 rows
+            if ($MaxRecords -ne 0 -and $MaxRecords -lt ($CurrentStartOffset + 4999)) { 
+                $CurrentEndOffset = $MaxRecords
+            } else {
+                $CurrentEndOffset = ($CurrentStartOffset + 4999)
+            }
+            Write-verbose "Querying SC for results: $CurrentStartOffset to $CurrentEndOffset" 
+            $query = @{
+                "tool"       = "vulnipdetail"
+                "sortField"  = "cveID"
+                "sortDir"    = "ASC"
+                "type"       = "vuln"
+                "sourceType" = "cumulative"
+                "query"      = @{
+                    "name"         = ""
+                    "description"  = ""
+                    "context"      = ""
+                    "status"       = "-1"
+                    "createdTime"  = 0
+                    "modifiedtime" = 0
+                    "sourceType"   = "cumulative"
+                    "sortDir"      = "desc"
+                    "tool"         = "listvuln"
+                    "groups"       = "[]"
                     "type"         = "vuln"
-                    "ispredefined" = $true
-                    "value"        = "$ID"
+                    "startOffset"  = $CurrentStartOffset
+                    "endOffset"    = $CurrentEndOffset
+                    "filters"      = [array]@{
+                        "id"           = "severity"
+                        "filterName"   = "severity"
+                        "operator"     = "="
+                        "type"         = "vuln"
+                        "ispredefined" = $true
+                        "value"        = "$ID"
+                    }
+                    "vulntool"     = "listvuln"
+                    "sortField"    = "severity"
                 }
-                "vulntool"     = "listvuln"
-                "sortField"    = "severity"
             }
-        }
 
-        $Splat = @{
-            Method   = "Post"
-            Body     = $(ConvertTo-Json $query -depth 5)
-            Endpoint = "/analysis"
-        }
+            $Splat = @{
+                Method   = "Post"
+                Body     = $(ConvertTo-Json $query -depth 5)
+                Endpoint = "/analysis"
+            }
+
+            $ThisResults = Invoke-PSTenableRest @Splat | Select-Object -ExpandProperty Response | Select-Object -ExpandProperty Results
+            if ($ThisResults) { #non zero records came back
+                $APIresults += $Thisresults
+                #move pagination line
+                $CurrentStartOffset = $CurrentEndOffset + 1
+            } 
+        } While ($ThisResults)

```

Comment:
Get back the results for this current page of query
if it is not empty add it to the array we are going to return to caller function
and move the pagination line 1 beyond the end of the last query

exit loop if the current results were empty (no more, error, whatever)